pulp_ansible/app/tasks/collections.py

```diff
@@ -1074,6 +1074,25 @@ async def run(self):
             pb.total = pb.done
 
 
+class DeclarativeNamespaceAvatar(DeclarativeArtifact):
+    """
+    Special handling for downloading Namespace Avatar Artifacts.
+    """
+
+    async def download(self):
+        """Allow download to fail, but not stop the sync."""
+        artifact_copy = self.artifact
+        try:
+            return await super().download()
+        except ClientError as e:
+            # Reset DA so that future stages can properly handle it
+            self.artifact = artifact_copy
+            self.deferred_download = True
```

Comment:
It reminds me a lot of https://github.com/pulp/pulp_deb/blob/main/pulp_deb/app/tasks/synchronizing.py#L192 with https://github.com/pulp/pulp_deb/blob/main/pulp_deb/app/tasks/synchronizing.py#L513 .

Can you elaborate more on the strategy? Do we handle the namespace as if no avatar existed, or do we turn the artifact into on-demand?

Reply:
It is very similar. The current strategy I am using is to handle the namespace as if no avatar existed. In the custom content saver stage I modify the namespace's metadata before save if the avatar failed to download.

Reply:
Probably, we could design it to accept a list of exceptions during download that would be failsafe.